### ISSUE TITLE
Question about CodeGenerator

### ISSUE THREAD

**Issue by FJays** [pixeltris/USharp#100]
Hello,

I created a class named `UUE_AnimTool`, the prefix `UUE` means "Unreal Extent"(The module name is `UnrealExtent`) with an `UObject` prefix.
When I generated the C# code, I found that the name of the class is `UE_AnimTool`, so I read the source code of `CodeGenerator.UpdateTypeNamePrefix`:

```c++
private string UpdateTypeNamePrefix(CodeGeneratorSettings.TypePrefix typePrefix, string str, 
    bool fullyQualifiedName, List<string> namespaces, string namespaceName = null)
{
    switch (typePrefix.Mode)
    {
        case CodeGeneratorSettings.TypePrefixMode.Strip:
            if (str.Length > 1 && str[0] == typePrefix.Char && char.IsUpper(str[1]))
            {
                str = str.Remove(0, 1);
            }
            break;
        case CodeGeneratorSettings.TypePrefixMode.Enforce:
            if (str.Length > 0)
            {
                if (str[0] != typePrefix.Char ||
                    (str.Length > 1 && char.IsUpper(str[0]) && char.IsLower(str[1])))
                {
                    str = typePrefix.Char + str;
                }
            }
            break;
    }
    return UpdateTypeNameNamespace(str, fullyQualifiedName, namespaces, namespaceName);
}
```

It compares `str[0]` and `typePrefix.Char`, but I don't know the reason. Why not insert the prefix dirrectly?

So, can you tell me?
Thanks~

**Comment by pixeltris** [comment]
Yea it's a flawed check. It mostly exists because enums always have a prefix in the path info e.g. `/Script/MyModule.EMyEnum` compared to other types here it's stripped such as `/Script/Engine.Actor` so I made a generic check which breaks easily. That check should only really be there for enum types.

**Comment by pixeltris** [comment]
This should now be fixed in https://github.com/pixeltris/USharp/commit/4c051761e7f23ab51139fc5bcc31152a85fb50ae. This commit also makes an additional change which forces you to use the prefixes in your C# defined classes as I think it's best to be as close to the C++ naming scheme as possible. Let me know if it still doesn't work as expected.

**Comment by FJays** [comment]
> This should now be fixed in [4c05176](https://github.com/pixeltris/USharp/commit/4c051761e7f23ab51139fc5bcc31152a85fb50ae). This commit also makes an additional change which forces you to use the prefixes in your C# defined classes as I think it's best to be as close to the C++ naming scheme as possible. Let me know if it still doesn't work as expected.

OK, thank you very much!
